setup.py

```diff
@@ -182,23 +140,42 @@ def members_appended(*ds):
         print('Warning: can not import pkgconfig python package.')
         pc = None
 
+    def lib_ext_kwargs(pc, prefix_env_var, lib_name, lib_pkg_name, pc_version, lib_subdir='lib'):
+        system_prefix = os.environ.get(prefix_env_var)
+        if system_prefix:
+            print(f'Detected and preferring {lib_pkg_name} [via {prefix_env_var}]')
+            return dict(include_dirs=[os.path.join(system_prefix, 'include')],
+                        library_dirs=[os.path.join(system_prefix, lib_subdir)],
+                        libraries=[lib_name])
+
+        if pc and pc.installed(lib_pkg_name, pc_version):
+            print(f'Detected and preferring {lib_pkg_name} [via pkg-config]')
+            return pc.parse(lib_pkg_name)
+        raise Exception(f'Could not find {lib_name} lib/headers, please set {prefix_env_var}')
```

Comment:
While we are at it:

setting the `BORG_*_PREFIX` env var is only one way.

maybe we should also suggest that maybe `PKG_CONFIG_PATH` path needs an additional entry to also find the `pkg-config` information (`*.pc` file) for the lib that was not found.

Reply:
Good idea. We can reuse the `lib_pkg_name` variable already used by pkg-config. So for openssl it would say:

`Exception: Could not find crypto lib/headers, please set BORG_OPENSSL_PREFIX or ensure libcrypto.pc is in PKG_CONFIG_PATH.`